### PR TITLE
linuxPackages: bump default 5.10 -> 5.15

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/system/boot/kernel.nix [hunk 1]
```diff
@@ -35,7 +35,7 @@ in
     };
 
     boot.kernelPackages = mkOption {
-      default = pkgs.linuxPackages;
+      default = if stdenv.isi686 then pkgs.linuxKernel.packages.linux_5_10 else pkgs.linuxPackages;
```
Comment (vcunat) [inline nixos/modules/system/boot/kernel.nix:38]: Oh, I originally thought that we'd define `linuxPackages` itself with an exception based on `stdenv.hostPlatform`, but I'm not really sure which approach to pick.

Comment (zowoq) [reply nixos/modules/system/boot/kernel.nix]: Don't mind how we do it. Not really sure how best to do what you're suggested though, all of the various `linuxPackages`, `linuxKernel`, etc packages and aliases is a bit of a mess.

**Comment by zowoq** [comment]
https://github.com/NixOS/nixpkgs/pull/152505#issuecomment-1005049314

Just noticed the comment about the `nfs3.simple` test, seems it does break with this.

File: nixos/modules/system/boot/kernel.nix (hunk 1 above)
Comment (vcunat) [reply nixos/modules/system/boot/kernel.nix]: I trust that there's only one place where the default kernel branch is defined, so my idea would mean
```diff
--- a/pkgs/top-level/linux-kernels.nix
+++ b/pkgs/top-level/linux-kernels.nix
@@ -530,3 +530,3 @@ in {
   packageAliases = {
-    linux_default = packages.linux_5_10;
+    linux_default = if stdenv.hostPlatform.isi686 then packages.linux_5_10 else packages.linux_5_15;
     # Update this when adding the newest kernel major version!
```

Comment (zowoq) [reply nixos/modules/system/boot/kernel.nix]: Done.

**Comment by jonringer** [comment]
I created a hydra jobset for you https://hydra.jonringer.us/jobset/nixpkgs/linux-5.15#tabs-evaluations

**Comment by jonringer** [comment]
tested jobset: https://hydra.jonringer.us/job/nixpkgs/linux-5.15/tested#tabs-constituents

**Comment by zowoq** [comment]
I had a quick look at the failing test, didn't see anything obvious (or I missed it). As `nfs` tests share code but only `3` breaks seems like it it is the kernel itself and not the test or an outdated dep.

**Comment by zowoq** [comment]
Guess we need to get this in soon so we don't run into next release freeze window.

Any objections to disabling the `nfs3.simple` test? I probably won't have time to look at it for a bit, if disabling it isn't an option someone else will need to try to fix it.

**Comment by zowoq** [comment]
`nfs3.simple` test is disabled, guess I'll merge this in ~24hrs?

**Review by jonringer — APPROVED** [review]

**Comment by zowoq** [comment]
PR to move `ena` / `amazonImage` to the 5.15 kernel: https://github.com/NixOS/nixpkgs/pull/172131

**Pull request merged**